test/test_simulation_build.jl

```diff
@@ -0,0 +1,339 @@
+path = joinpath(pwd(), "test_sequence_build")
+!isdir(path) && mkdir(path)
+
+function test_sequence_build(file_path::String)
+    stages_definition = Dict(
+        "UC" => Stage(GenericOpProblem, template_basic_uc, c_sys5_uc, GLPK_optimizer),
+        "ED" => Stage(GenericOpProblem, template_ed, c_sys5_ed, GLPK_optimizer),
+    )
+
+    sequence = SimulationSequence(
+        step_resolution = Hour(24),
+        order = Dict(1 => "UC", 2 => "ED"),
+        feedforward_chronologies = Dict(("UC" => "ED") => Synchronize(periods = 24)),
+        horizons = Dict("UC" => 24, "ED" => 12),
+        intervals = Dict(
+            "UC" => (Hour(24), Consecutive()),
+            "ED" => (Hour(1), Consecutive()),
+        ),
+        feedforward = Dict(
+            ("ED", :devices, :Generators) => SemiContinuousFF(
+                binary_from_stage = PSI.ON,
+                affected_variables = [PSI.ACTIVE_POWER],
+            ),
+        ),
+        ini_cond_chronology = InterStageChronology(),
+    )
+
+    @testset "Test Simulation Simulation Sequence Validation" begin
+        @test length(findall(x -> x == 2, sequence.execution_order)) == 24
+        @test length(findall(x -> x == 1, sequence.execution_order)) == 1
+    end
+
+    @testset "Simulation with provided initial time" begin
+        second_day = DayAhead[24] + Hour(1)
+        sim = Simulation(
+            name = "test",
+            steps = 1,
+            stages = stages_definition,
+            stages_sequence = sequence,
+            simulation_folder = file_path,
+            initial_time = second_day,
```

Comment:
This test does not catch the condition where a system has a single long forecast.  The RTS errors:
```julia
julia> sim = Simulation(
           name = "aggregation",
           steps = 2,
           stages = stages_definition,
           stages_sequence = sequence,
           simulation_folder = file_path,
           initial_time = Dates.DateTime("2020-01-05T00:00:00")
           )
Simulation()


julia> build!(sim)
┌ Debug: (3600000 milliseconds, 3600000 milliseconds)
└ @ PowerSimulations ~/Documents/repos/PowerSimulations.jl/src/core/feedforward.jl:38
[ Info: Building Stage 2-ED
┌ Debug: Instantiating the JuMP model
└ @ PowerSimulations ~/Documents/repos/PowerSimulations.jl/src/core/psi_container.jl:18
┌ Debug: Building HydroDispatch with HydroDispatchRunOfRiver formulation
└ @ PowerSimulations ~/Documents/repos/PowerSimulations.jl/src/core/operations_problem.jl:452
┌ Debug: assign_variable
│   name = :P_HydroDispatch
└ @ PowerSimulations ~/Documents/repos/PowerSimulations.jl/src/core/psi_container.jl:331
┌ Debug: 2020-01-05T00:00:00
└ @ PowerSimulations ~/Documents/repos/PowerSimulations.jl/src/devices_models/devices/hydro_generation.jl:176
ERROR: ArgumentError: forecast InfrastructureSystems.ForecastKey(InfrastructureSystems.DeterministicInternal, 2020-01-05T00:00:00, "get_rating") is not stored
Stacktrace
```

Reply:
@claytonpbarrows this is resolved via 
55a4799